src/app/transactions/MempoolFeeStats.tsx

```diff
@@ -1,127 +1,307 @@
-import { Caption } from '../../ui/typography';
 import { ExplorerErrorBoundary } from '../_components/ErrorBoundary';
-import { StatSection } from '../_components/Stats/StatSection';
-import { Wrapper } from '../_components/Stats/Wrapper';
+import {
+  TransactionTypeFilterMenu,
+  TransactionTypeFilterTypes,
+  mapTransactionTypeToFilterValue,
+} from '../_components/Stats/TransactionTypeFilterMenu';
+import { MempoolFeePieChart, getTxTypePieChartColor } from './MempoolFeePieChart';
 
-function MempoolFeeByTxType({
-  mempoolFeeTokenTransfer,
-  mempoolFeeContractCall,
-  mempoolFeeSmartContract,
-}: {
-  mempoolFeeTokenTransfer: number;
-  mempoolFeeContractCall: number;
-  mempoolFeeSmartContract: number;
-}) {
-  return (
-    <HStack divider={<Caption color={'border'}>|</Caption>} gap={1}>
-      <Tooltip
-        label={`Token transfer tx fee: ${mempoolFeeTokenTransfer / MICROSTACKS_IN_STACKS} STX`}
-      >
-        <Flex gap={0.5} alignItems={'center'} justifyContent={'center'}>
-          <Icon as={getTxTypeIcon('token_transfer')} />
-          <Box suppressHydrationWarning>
-            {`${Number((mempoolFeeTokenTransfer / MICROSTACKS_IN_STACKS).toFixed(3))}`} STX
-          </Box>
-        </Flex>
-      </Tooltip>
-      <Tooltip
-        label={`Contract call tx fee: ${mempoolFeeContractCall / MICROSTACKS_IN_STACKS} STX`}
-      >
-        <Flex gap={0.5} alignItems={'center'} justifyContent={'center'}>
-          <Icon as={getTxTypeIcon('contract_call')} size={3} />
-          <Box suppressHydrationWarning>
-            {`${Number((mempoolFeeContractCall / MICROSTACKS_IN_STACKS).toFixed(3))}`} STX
-          </Box>
-        </Flex>
-      </Tooltip>
-      <Tooltip
-        label={`Smart contract tx fee: ${mempoolFeeSmartContract / MICROSTACKS_IN_STACKS} STX`}
-      >
-        <Flex gap={0.5} alignItems={'center'} justifyContent={'center'}>
-          <Icon as={getTxTypeIcon('smart_contract')} size={3} />
-          <Box suppressHydrationWarning>
-            {`${Number((mempoolFeeSmartContract / MICROSTACKS_IN_STACKS).toFixed(3))}`} STX
-          </Box>
-        </Flex>
-      </Tooltip>
-    </HStack>
-  );
-}
+export const getFeePriorityIcon = (priority: keyof MempoolFeePrioritiesAll) => {
+  switch (priority) {
+    case 'no_priority':
+      return <Icon as={TbCircleMinus} size={4} color="slate.600" />;
+    case 'low_priority':
+      return <Icon as={TbCircleChevronDown} size={4} color="red.600" />;
+    case 'medium_priority':
+      return <Icon as={TbCircleChevronUp} size={4} color="orange.600" />;
+    case 'high_priority':
+      return <Icon as={TbCircleChevronsUp} size={4} color="green.600" />;
+    default:
+      throw new Error('Invalid priority');
+  }
+};
 
-function MempoolFeeSection({
+function MempoolFeePrioritCard({
   mempoolFeeResponse,
   priority,
   stxPrice,
-  ...rest
+  txTypeFilter,
 }: {
-  mempoolFeeResponse: MempoolFeePriorities;
+  mempoolFeeResponse: MempoolFeePriorities; // TODO:
```

Comment:
Could you please add more details to the todo comment?

Reply:
Just removing as I don't remember what the TODO was for